python/lsst/obs/pfs/isrTask.py

```diff
@@ -68,6 +73,66 @@ class PfsIsrTaskConfig(ipIsr.IsrTaskConfig):
 
     def validate(self):
         super().validate()
+        if not self.doSaturationInterpolation and "SAT" in self.maskListToInterpolate:  # fixed on LSST master
+            self.maskListToInterpolate.remove("SAT")
+#
+# Code to handle the broken shutter matrix matrix inverse cache
+#
+
+
+class BrokenShutterKernelCache:
+    __cache = {}
+    __maxsize = 0
+
+    def __init__(self, maxsize=0):
+        self.setCacheSize(maxsize)
+
+    @classmethod
+    def compute(cls, kern, t_exp):
+        key = int(t_exp + 0.5)
+        if key not in cls.__cache:
+            if cls.__maxsize > 0 and len(cls.__cache) >= cls.__maxsize:  # need to delete an old kernel
+                k_min = None                                            # the least popular kernel
+                n_min = 0
+                for k, v in cls.__cache.items():
+                    if k_min is None or v[1] < n_min:
+                        k_min = k
+                        n_min = v[1]
+
+                if k_min is not None:
+                    print(f"Clearing cached kernel for {k_min}")
+                    del cls.__cache[k_min]
+
+            print(f"Computing ikern for {key} ({t_exp}s)")
```

Comment:
Can the ``print`` call be converted to a ``log.info`` call? Similar occurrences later in the class (except the ``show()`` method which is fine as-is)

Reply:
Well, not trivially as the cache has no associated log or parent, so it'd be stand-alone --- that's why I didn't do it with logging in the first place.  But sure, I can see how hard that would be to implement.